File: movieApp/app/src/main/java/com/lbc/practice/movieapp/model/resource/MovieRepository.java
package com.lbc.practice.movieapp.model.resource;

import android.content.Context;


import javax.inject.Inject;

public class MovieRepository implements MovieDataSource {



     MovieDataSource movieRemoteDataSource;
     MovieDataSource movieLocalDataSource;

    @Inject
    public MovieRepository(@RemoteScope MovieDataSource movieRemoteDataSource, @LocalScope MovieDataSource movieLocalDataSource){
        this.movieRemoteDataSource = movieRemoteDataSource;
        this.movieLocalDataSource = movieLocalDataSource;
    }


//    @Inject
//    MovieDataSource movieRemoteDataSource;
//
//    @Inject
//    MovieDataSource movieLocalDataSource;
//
//    public MovieRepository() {
//
//        MovieRepositoryComponent movieRepositoryComponent = DaggerMovieRepositoryComponent.builder().movieRepositoryModule(new MovieRepositoryModule()).build();
//        movieRepositoryComponent.repositoryInject(this);
//    }

//    private static MovieRepository movieRepository;


//    public static MovieRepository getInstance(){
//        if (movieRepository ==null) {
//            movieRepository =new MovieRepository();
//        }
//        return movieRepository;
//    }

//    public static void setMovieRepository(MovieRepository movieRepository) {
//        MovieRepository.movieRepository = movieRepository;
//    }

    @Override
    public void seeMovieInfo(String apiKey, String language, int page, LoadDataCallBack loadDataCallBack) {
        movieRemoteDataSource.seeMovieInfo(apiKey,language,page,loadDataCallBack);
    }

    @Override
    public void favoriteMovieInfo(Context context, LoadDataCallBack loadDataCallBack) {
        movieLocalDataSource.favoriteMovieInfo(context,loadDataCallBack);
    }


}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/data/resource/MovieRepository.kt
package com.lbc.practice.movieapp.data.resource;

import android.content.Context;

import com.lbc.practice.movieapp.data.resource.local.MovieLocalDataSource;
import com.lbc.practice.movieapp.data.resource.remote.MovieRemoteDataSource;
import javax.inject.Inject


class MovieRepository : MovieDataSource {


    var movieRemoteDataSource: MovieDataSource ?= null
    var movieLocalDataSource: MovieDataSource ?= null

    @Inject
    constructor(@RemoteScope movieRemoteDataSource: MovieDataSource?, @LocalScope movieLocalDataSource: MovieDataSource?) {
        this.movieRemoteDataSource = movieRemoteDataSource
        this.movieLocalDataSource = movieLocalDataSource
    }


    override fun seeMovieInfo(apiKey: String, language: String, page: Int, loadDataCallBack: MovieDataSource.LoadDataCallBack) {
        movieRemoteDataSource!!.seeMovieInfo(apiKey, language, page, loadDataCallBack)
    }

    override fun favoriteMovieInfo(context: Context, loadDataCallBack: MovieDataSource.LoadDataCallBack) {
        movieLocalDataSource!!.favoriteMovieInfo(context, loadDataCallBack)
    }

}
